google-account-plugin/src/com/google/cloud/tools/intellij/stats/GoogleUsageTracker.java

```diff
@@ -153,11 +152,14 @@ public void sendEvent(
         // I think 'virtual' indicates these don't correspond to real web pages.
         postData.add(new BasicNameValuePair(IS_VIRTUAL_KEY, STRING_TRUE_VALUE));
         String fullMetadataString = systemMetadataKeyValues;
-        if (eventLabel != null) {
-          // Not using ImmutableMap.of() because I want to handle null for values.
-          Map<String, Integer> eventMetadataMap = Maps.newHashMap();
-          eventMetadataMap.put(METADATA_ESCAPER.escape(eventLabel), eventValue);
-          fullMetadataString = fullMetadataString + "," + METADATA_JOINER.join(eventMetadataMap);
+        if (metadataMap != null && !metadataMap.isEmpty()) {
+          Map<String, String> escapedMap = metadataMap.entrySet()
+              .stream()
+              .collect(Collectors.toMap(
```

Comment:
with map() we'd still need to collect() the values so I believe this is more concise.